src/managers/cluster.py

```diff
@@ -85,3 +88,67 @@ def update_credentials(self, username: str, password: str) -> None:
             client.update_password(username=username, new_password=password)
         except EtcdUserManagementError:
             raise
+
+    def add_member(self, unit_name: str) -> None:
+        """Add a new member to the etcd cluster."""
+        # retrieve the member information for the newly joined unit from the set of EtcdServers
+        member_name = ""
```

Comment:
nit, I believe we can be a bit more concise here and get rid of these 3 variables and quite a bit of indentation:
```
server = next(iter([s for s in self.state.servers if s.unit_name == unit_name]), None)
if not server:
    raise KeyError(f"Peer relation data for unit {unit_name} not found.")

....
```